wgsl/index.bs

```diff
@@ -689,84 +682,240 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
 The terms defined in this section express counts of 8-bit bytes.
 
 We will use the following notation:
+* <dfn noexport>AlignOf</dfn>(|T|) is the alignment of host-shareable type |T|.
+* AlignOf(|S|, |M|) is the alignment of member |M| of the host-shareable structure |S|.
+* <dfn noexport>SizeOf</dfn>(|T|) is the size of host-shareable type |T|.
+* SizeOf(|S|, |M|) is the size of member |M| of the host-shareable structure |S|.
+* <dfn noexport>StrideOf</dfn>(|A|) is the element stride of host-shareable array type |A|.
+* <dfn noexport>OffsetOf</dfn>(|S|, |M|) is the offset of member |M| from the start of the host-shareable structure |S|.
+
+
+#### Type Alignment and Size ####  {#type-alignment-and-size}
+
+Every [=host-shareable=] data type has an alignment and size value.
+
+The alignment of a type specifies the byte offsets at which a value of that type
+may be placed relative to the start of the host-shareable structure or array.
+
+The size of a type is the number of contiguous bytes reserved in
+host-shareable memory for the purpose of storing a value of the type.
+The size may include non-addressable padding at the end of the type.
+Consequently, loads and stores of a value might access fewer memory locations
+than value's size.
+
+Both are defined recursively in the following table:
```

Comment:
`The alignment and size of all host-shareable types are defined recursively in the following table:`

Reply:
Done